### PR TITLE
Update guide for async functions

### PR DESCRIPTION
**Pull request by omelancon**
Fixed a few typos on the way

Replaced function name do by foo because do is a reserved keyword

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -183,27 +183,34 @@ const foo (a = 0) =>  bar(a)
...
 // good
-const do = function* () { ... }
+const foo = function* () { ... }
+
+// async functions
+// bad
+async function foo () { ... }
+
+// good
+const foo = async () { ... }
```
Comment (omelancon) [inline README.md]: That here is the main change

File: README.md [hunk 2]
```diff
@@ -183,27 +183,34 @@ const foo (a = 0) =>  bar(a)
...
 // good
-const do = function* () { ... }
+const foo = function* () { ... }
+
+// async functions
+// bad
+async function foo () { ... }
+
+// good
+const foo = async function () { ... }
```
Comment (omelancon) [inline README.md]: That here is the focus of this PR

Comment (pre63) [inline README.md]: actually according to 3.5 it should be 
```
const foo = async () => { ... }
```

File: README.md [hunk 3]
```diff
@@ -565,6 +572,7 @@ const state => {
 // good
 const state => Object.assign({}, state, { id : 1 }) // native js
 
+// best
```
Comment (pre63) [inline README.md]: best is sort of subjective, I'd rather we stop using it, `good` works

**Pull request merged**